src/zalopay/services/zod.ts

```diff
@@ -0,0 +1,104 @@
+import getJsonResponse from "../helpers/getJsonResponse";
+import {
+  ObjectSerializer,
+  ZODCreateInvoiceRequest,
+  ZODCreateInvoiceResponse,
+  ZODQueryInvoiceResponse,
+  ZODQueryStatusResponse
+} from "../models/models";
+import Service from "../service";
+import HmacUtils from "../utils/hmacUtils";
+import { ZaloPayClient } from "../zaloPayClient";
+import ZODResource from "./resource/zodResource";
+
+class ZOD extends Service {
+  private readonly _create: ZODResource;
+  private readonly _query_invoice: ZODResource;
+  private readonly _query_status: ZODResource;
+
+  private hmacUtils: HmacUtils;
+
+  public constructor(client: ZaloPayClient) {
+    client.httpClient.defaults.baseURL =
+      client.config.env === "sandbox"
```

Comment:
I don't think so,
The user only needs to set the env variable which is sandbox or production. And the endpoint should be handled by us.
I put this logic here because the default base URL is "https://sb-openapi.zalopay.vn". The ZOD service has its base Url so we override the default URL here.